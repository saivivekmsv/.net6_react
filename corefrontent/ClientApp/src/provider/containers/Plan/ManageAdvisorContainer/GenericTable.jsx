import React, { useState } from "react";
import { Link } from "react-router-dom";
import { CsplTable as Table } from "../../../components";
import AddToolTip from "../../../components/AddToolTip";
import { MANAGE_PLAN_ROUTES } from "../../../utils";

const columns = [
  {
    label: "Advisor Company Name",
    className: "column-custodianName",
    columnName: "advisorName",
    isSortable: true,
    orderBy: "",
  },
  {
    label: "E Mail",
    className: "column-custodianMail",
    columnName: "advisorMail",
    isSortable: true,
    orderBy: "",
  },
  {
    label: "Phone no.",
    className: "column-custodianPhone",
    columnName: "advisorPhone",
    isSortable: true,
    orderBy: "",
  },
];

const GenericTable = ({
  data = [],
  isLoading,
  handleSort,
  totalRecords,
  scrollEndCallBack,
}) => {
  const [sortingData, setSortingData] = useState({});

  const onSortClick = (item, index, orderBy) => {
    const details = {
      column: item.columnName,
      columnIndex: index,
      orderBy,
    };
    setSortingData(details);
    handleSort(details);
  };

  const scrollCallBack = (_, pageDetails) => {
    scrollEndCallBack(pageDetails, "generic");
  };

  return (
    <Table isLoading={isLoading}>
      <Table.Thead>
        <Table.Tr>
          {columns.map((item, index) => {
            const orderBy =
              sortingData.columnIndex === index ? sortingData.orderBy : "";
            return (
              <Table.Th
                key={index}
                className={item.className}
                isSortable={item.isSortable}
                orderBy={orderBy}
                onClick={(sortOrder) => onSortClick(item, index, sortOrder)}
              >
                {item.label}
              </Table.Th>
            );
          })}
        </Table.Tr>
      </Table.Thead>
      <Table.Tbody
        scrollEndCallBack={scrollCallBack}
        totalRecords={totalRecords}
      >
        {data.map((advisor, index) => {
          return (
            <Table.Tr key={index}>
              <Table.Td className="column-custodianName">
                <Link
                  to={`${MANAGE_PLAN_ROUTES.ADD_ADVISOR}/edit/${advisor.id}`}
                >
                  <AddToolTip name={advisor.advisorName} />
                </Link>
              </Table.Td>
              <Table.Td className="column-custodianMail">
                <AddToolTip name={advisor.advisorMail} />
              </Table.Td>
              <Table.Td className="column-custodianPhone">
                {advisor.advisorPhone}
              </Table.Td>
            </Table.Tr>
          );
        })}
      </Table.Tbody>
    </Table>
  );
};

export default GenericTable;
