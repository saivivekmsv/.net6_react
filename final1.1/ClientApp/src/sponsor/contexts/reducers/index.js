export * from "./app-layout";
export * from "./manage-company";
export * from "./create-plan";
export * from "./manage-eligibility";
export * from "./manage-census";
export * from "./manage-payroll";
export * from "./manage-maintenance";
export * from "./manage-report";
export * from "./manage-mapper";
