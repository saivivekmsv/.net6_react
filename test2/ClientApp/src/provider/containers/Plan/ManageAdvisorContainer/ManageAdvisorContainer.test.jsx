import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import ManageAdvisorContainer from "./ManageAdvisorContainer";
import { CreatePlanProvider } from "../../../contexts";

describe("ManageAdvisorContainer", () => {
  const props = {};
  test("to match snapshot active", () => {
    const { baseElement } = render(
      <CreatePlanProvider>
        <Router>
          <ManageAdvisorContainer {...props} />
        </Router>
      </CreatePlanProvider>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
