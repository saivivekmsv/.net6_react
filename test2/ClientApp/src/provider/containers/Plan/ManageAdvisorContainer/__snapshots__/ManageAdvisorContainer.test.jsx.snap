// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageAdvisorContainer to match snapshot active 1`] = `
<body>
  <div>
    <div
      class="d-flex flex-column manage-company h-100"
    >
      <div
        class="d-flex layout-header align-items-center"
      >
        <div>
          <nav
            aria-label="breadcrumb"
          >
            <ol
              class="breadcrumb"
            >
              <li
                class="breadcrumb-item"
              >
                <a
                  href="#"
                  role="button"
                >
                  <a
                    href="/home"
                  >
                    Home
                  </a>
                </a>
              </li>
              <li
                class="breadcrumb-item"
              >
                <a
                  href="#"
                  role="button"
                >
                  <a
                    href="/plans"
                  >
                    Manage Plan
                  </a>
                </a>
              </li>
            </ol>
          </nav>
          <h5 />
        </div>
        <div
          class="d-flex align-items-center justify-content-center tab-wrapper"
        />
        <div
          class="ml-auto"
        >
          <div />
        </div>
      </div>
      <div
        class="layout-content h-100 overflow-auto"
      >
        <div
          class="d-flex flex-column"
        >
          <div
            class="plan-heading mb-10"
          >
            Advisor Information
          </div>
          <div
            class="loader-wrapper "
          >
            <div
              class="d-flex flex-column cspl-table undefined"
            >
              <div
                class="d-flex table-head align-items-center undefined"
              >
                <div
                  class="d-flex table-row undefined"
                >
                  <div
                    class="d-flex table-head-cell column-custodianName sortable "
                  >
                    Company Name
                  </div>
                  <div
                    class="d-flex table-head-cell column-custodianMail sortable "
                  >
                    E Mail
                  </div>
                  <div
                    class="d-flex table-head-cell column-custodianPhone sortable "
                  >
                    Phone No.
                  </div>
                </div>
              </div>
              <div
                class="d-flex flex-column table-body undefined"
              >
                <div
                  class="d-flex table-row undefined"
                >
                  <div
                    class="d-flex table-cell column-custodianName"
                  >
                    <a
                      href="/manage-plan/add-advisor/edit/undefined"
                    >
                      MEP -102
                    </a>
                  </div>
                  <div
                    class="d-flex table-cell column-custodianMail"
                  >
                    advisor@mep207.com
                  </div>
                  <div
                    class="d-flex table-cell column-custodianPhone"
                  >
                    2092889207
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
